### ISSUE TITLE
Bug related to nested mapped types

### ISSUE THREAD

**Issue by hediet** [Microsoft/TypeScript#15638]
**TypeScript Version:**  2.3 (Playground)

**Code**

```ts
type MyType = { prop1: string, prop2: number };

type OneOf<T> = T[keyof T];
type Wrap<TType extends object> = {[TName in keyof TType]: {[TName2 in TName]: TType[TName]}};
type OneOf_Wrap<TType extends object> = OneOf<Wrap<TType>>;

type A = OneOf<Wrap<MyType>>
type B = OneOf_Wrap<MyType>;

let a: A; // language service says:  type A = { prop1: string; } | { prop2: number; }
let b: B;  // language service says:  type B = { prop1: string | number; prop2: string | number; }
a = b;
```

**Expected behavior:**
No compile error, types A and B are structurally equal.

**Actual behavior:**
The following error is reported for `a = b;`:
```
Type '{ prop1: string | number; prop2: string | number; }' is not assignable to type '{ prop1: string; } | { prop2: number; }'.
  Type '{ prop1: string | number; prop2: string | number; }' is not assignable to type '{ prop2: number; }'.
    Types of property 'prop2' are incompatible.
      Type 'string | number' is not assignable to type 'number'.
        Type 'string' is not assignable to type 'number'.
```

**Comment by hediet** [comment]
Is fixed in 3.4.0.
